### PR TITLE
Add build_args also to `ct_s2i_build_as_df` function.

### PR DESCRIPTION
**Pull request by phracek**
Add build_args also to `ct_s2i_build_as_df` function.
We have to functions to build image. This one and `ct_test_app_dockerfile`

In case of --ulimit is defined in s2i_args variable we will add this parameter to docker/podman command line.

This pull request blocks https://github.com/sclorg/s2i-nodejs-container/pull/425.

### CODE REVIEW

**Comment by phracek** [comment]
[test]

File: test-lib.sh [hunk 1]
```diff
@@ -988,8 +989,9 @@ EOF
     # Check if -v parameter is present in s2i_args and add it into docker build command
     read -ra mount_options <<< "$(echo "$s2i_args" | grep -o -e '\(-v\)[[:space:]]\.*\S*' || true)"
 
+    echo "$s2i_args" | grep -q '\--ulimit' && build_args=$(echo "$s2i_args" | grep -o -e '\--ulimit[[:space:]]\S*\w=*')
```
Comment (zmiklank) [inline test-lib.sh:992]: What is purpose of this line? Can you make a comment about it in the code please?

Comment (phracek) [reply test-lib.sh:992]: @zmiklank It gets s2i-args and parse if `--ulimit` is present. If so then get all `--ulimit parameters` a pass them into docker/podman build command.
But the comment is correct.

Comment (phracek) [reply test-lib.sh:994]: @zmiklank The comment is added into code and also into commit message https://github.com/sclorg/container-common-scripts/pull/368/commits/cb248e0eb51c785f812d432820d64f178c3eedad

**Comment by phracek** [comment]
[test]

**Comment by phracek** [comment]
[test]

File: test-lib.sh (hunk 1 above)
Comment (phracek) [reply test-lib.sh:994]: Yes, but which one? There are a lot of them. For case of `s2i-nodejs-container` there is ```--ulimit nofile=4096:4096```. Passing build parameters to function `ct_s2i_build_as_df` is done at the end.

In this case, we should define a new variable and specify it, like --build-args, and the rest of the parameters will be used by podman. But this is used only by one specific container. Does it make sense either?

Comment (zmiklank) [reply test-lib.sh:994]: What if we pass build args to the function like this?
```
diff --git a/test-lib.sh b/test-lib.sh
index c38647e..c0be7ab 100644
--- a/test-lib.sh
+++ b/test-lib.sh
@@ -880,6 +880,17 @@ ct_s2i_usage()
     docker run --rm "$img_name" bash -c "$usage_command"
 }
 
+ct_s2i_build_as_df()
+{
+    local app_path=$1; shift
+    local src_image=$1; shift
+    local dst_image=$1; shift
+    local s2i_args="$*";
+
+    ct_s2i_build_as_df_build_args $app_path $src_image $dst_image "" $s2i_args
+
+}
+
 # ct_s2i_build_as_df APP_PATH SRC_IMAGE DST_IMAGE [S2I_ARGS]
 # ----------------------------
 # Create a new s2i app image from local sources in a similar way as source-to-image would have used.
@@ -888,11 +899,12 @@ ct_s2i_usage()
 # Argument: DST_IMAGE - image name to be used during the tagging of the s2i build result
 # Argument: S2I_ARGS - Additional list of source-to-image arguments.
 #                      Only used to check for pull-policy=never and environment variable definitions.
-ct_s2i_build_as_df()
+ct_s2i_build_as_df_build_args()
 {
     local app_path=$1; shift
     local src_image=$1; shift
     local dst_image=$1; shift
+    local build_args=$1; shift
     local s2i_args="$*";
     local local_app=upload/src/
     local local_scripts=upload/scripts/
```

Comment (phracek) [reply test-lib.sh]: There is a typo. I guess the argument "" is empty and it will not work. I have to clarify it by real tests. Give me mmnt.

**Comment by phracek** [comment]
I have tested this updated PR with s2i-nodejs-container repo and it works

[test]

File: test-lib.sh [hunk 2]
```diff
@@ -894,6 +894,26 @@ ct_s2i_build_as_df()
     local src_image=$1; shift
     local dst_image=$1; shift
     local s2i_args="$*";
+
+    ct_s2i_build_as_df_build_args "$app_path" "$src_image" "$dst_image" "" "$s2i_args"
+}
+
+# ct_s2i_build_as_df_build_args APP_PATH SRC_IMAGE DST_IMAGE [S2I_ARGS]
```
Comment (zmiklank) [inline test-lib.sh:901]:
```
-  # ct_s2i_build_as_df_build_args APP_PATH SRC_IMAGE DST_IMAGE [S2I_ARGS]
+  # ct_s2i_build_as_df_build_args APP_PATH SRC_IMAGE DST_IMAGE BUILD_ARGS [S2I_ARGS]
```

Just a nitpick in comment.

Also could you please explain in comment of `ct_s2i_build_as_df` function why we created the wrapper?

Comment (phracek) [reply test-lib.sh]: It should be explained by https://github.com/sclorg/container-common-scripts/pull/368/commits/ca4f716c7b500c2ad9796a7c79b94f42e9848474

**Comment by phracek** [comment]
[test]

**Review by zmiklank — APPROVED** [review]

**Pull request merged**